test/unit/abilities/provider_any_test.rb

```diff
@@ -69,6 +69,47 @@ def test_policies_not_tenant
       assert_cannot ability, :show, Accounts::AccountCreatedEvent.create(account, user)
     end
 
+    class ShowAlertRelatedEventTest < ProviderAnyTest
+      setup do
+        user.stubs(:has_permission?)
+
+        cinstance = FactoryBot.build_stubbed(:simple_cinstance)
+        alert = FactoryBot.build_stubbed(:limit_violation, cinstance: cinstance)
+        @limit_violation_reached_provider_event = Alerts::LimitViolationReachedProviderEvent.create(alert)
+      end
+
+      attr_reader :limit_violation_reached_provider_event
+
+      test 'cannot show AlertRelatedEvent when user does not have :monitoring' do
+        user.expects(:has_permission?).with(:monitoring).returns(false)
+
+        assert_cannot ability, :show, limit_violation_reached_provider_event
+      end
+
+      test 'can show AlertRelatedEvent when user has :monitoring and the user has no services (old permissions)' do
```

Comment:
Do you mean `:service_permissions` rolling update? Maybe it would be nice to add a comment about it.

Reply:
The "old permissions" part came from [this comment](https://github.com/3scale/porta/blob/master/app/models/user/permissions.rb#L96-L98) 😬 . What comment do you think would be good in here?
Something like "Check User::Permissions#has_access_to_all_services?"?

Reply:
Thanks @thalesmiguel ! 

I checked this further, and it does seem to be related to the `:service_permissions` rolling update.

So, the test description is not really correct then. I think it should say something like (maybe it's too long!):
```
can show AlertRelatedEvent when user has :monitoring and the user has access to all services through old permission system
```

So, it's not that the user does not have service, it's that the member permissions of the user do not include the `services` section, and this happens for users before `:service_permissions` rolling update was introduced.

Maybe you can reference this rolling update in the comment so it is clearer.

Reply:
Thanks for the explanation, @mayorova!
And nothing wrong with long names if it makes life easier 😬 
Updated in 972db04.